src/battle_script_commands.c

```diff
@@ -17236,3 +17248,76 @@ void BS_TryRevivalBlessing(void)
         MarkBattlerForControllerExec(gBattlerAttacker);
     }
 }
+
+void BS_CheckPokeFlute(void)
+{
+    NATIVE_ARGS();
+    u8 battler;
+    s32 i;
+    u32 monToCheck, status;
+    u16 species;
+    u8 abilityNum;
+    gBattleCommunication[MULTISTRING_CHOOSER] = 0;
+    monToCheck = 0;
```

Comment:
I understand that this is a direct port from FRLG, but the code honestly seems pretty choppy. I think it'd be better if we could rebuild it from the ground up in a more efficient manner.

Reply:
Not entirely sure what I can do differently here do to BtlController_EmitSetMonData requiring me to do the loop twice for each side

Reply:
Then rather than near-duplicate code with identical conditions we should write a macro/function to deal with it in a more centralised fashion imo

Reply:
Right, added a function for it then. Should be good now